### PR TITLE
feat(*): Add new round logic and end round dialog

### PR DESCRIPTION
**Pull request by dhruvd-grappus**
* Fix get random word
* Save id of last drawing player
* Pause timer when round ends (show dialog of correct word in UI) then start next round

https://github.com/noorbhatia10/grappus_skribbl/assets/110366216/3e0cc488-c6ad-494e-a401-a734eee1baf0

### CODE REVIEW

File: api/lib/session/bloc/session_bloc.dart [hunk 1]
```diff
@@ -49,9 +48,6 @@ class SessionBloc extends BroadcastBloc<SessionEvent, SessionState> {
         players: players,
         eventType: EventType.addPlayer,
         points: state.points,
-
-        ///Making [currentPlayerId] null so that it does not override
-        currentPlayerId: null,
```
Comment (noorbhatia10) [inline api/lib/session/bloc/session_bloc.dart]: We should make currentPlayerId null after the player is added

File: lib/views/game/cubit/game_cubit.dart [hunk 2]
```diff
@@ -20,16 +20,19 @@ class GameCubit extends Cubit<GameState> {
   StreamSubscription<SessionState?>? _sessionStateSub;
 
   Future<void> connect(String name) async {
+    final playerID = const Uuid().v4();
```
Comment (noorbhatia10) [inline lib/views/game/cubit/game_cubit.dart:23]: Did you try assigning uid on backend?

File: lib/views/game/view/round_end_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+              ],
+            ),
+          ),
+        ),
+      ),
+      listener: (context, state) {
+        if (remainingSeconds == 0) {
+          _tickerSub.cancel();
+          Navigator.of(context).pop();
+        }
```
Comment (noorbhatia10) [inline lib/views/game/view/round_end_dialog.dart]: What happens if there's a lag in the server? How can we ensure that dialog closes at the right time when new round is about to start?

File: lib/views/game/cubit/game_cubit.dart (hunk 2 above)
Comment (dhruvd-grappus) [reply lib/views/game/cubit/game_cubit.dart:23]: Will target in the next PR, heres my approach:
- Create a REST API called `/connect?name=test` which generates and returns the `playerID` to the client and adds a `Player` in the `SessionBloc`
- Then connect to the websocket using that uid and name on client side

If we tried to send a uid back to client via websocket (`currentPlayerID` rn), we wouldnt be sure which client would the ID go to (say 10 players connecting at the same time)

**Review by noorbhatia10 — APPROVED** [review]

**Pull request merged**